pdns/dnsparser.cc

```diff
@@ -96,8 +96,8 @@ shared_ptr<DNSRecordContent> DNSRecordContent::deserialize(const DNSName& qname,
 
   memcpy(&packet[0], &dnsheader, sizeof(dnsheader)); pos+=sizeof(dnsheader);
 
-  char tmp[6]="\x0" "\x0\x1" "\x0\x1"; // root question for ns_t_a
-  memcpy(&packet[pos], &tmp, 5); pos+=5;
+  const std::array<uint8_t, 6> tmp= {'\x0', '\x0', '\x1', '\x0', '\x1', '\0' }; // root question for ns_t_a
+  memcpy(&packet[pos], tmp.data(), 5); pos+=5;
```

Comment:
Same on this one I can make an other PR, asked myself if a `string_view` is not better.

Reply:
Some question here: what's the reason you suggest string_view instead of just string?

Reply:
Would have done something like:
```cpp
std::string_view tmp = "\x0" "\x0\x1" "\x0\x1\0"; // root question for ns_t_a
memcpy(&packet[pos], tmp.data(), tmp.length() - 1); pos += 5;
```
Or 
```cpp
std::string_view tmp = "\x0" "\x0\x1" "\x0\x1"; // root question for ns_t_a
memcpy(&packet[pos], tmp.data(), tmp.length()); pos += 5;
```

Reply:
Why would I do that: I mostly just want a contiguous view on constant Read only memory as `src` for `memcpy` but if `string` is fine I can just go `const string`. But what drived me to `string_view` is that `string` is not usable in `constexp` and that I just want a view of memory like a `char *` not a heap allocated thing. :)

Reply:
I think a `std::array` is fine in this case, but please fix the 5 vs 6 inconsistency.

Reply:
That's a good point: I prefers to spell it out, I added did a size of 6 to explicitly include a `\0` like the `char *` did but I suppose it's just better do "do not add this `\0` not used. and go for 5.

To avoid inconsistency: keep 5 and do not include `\0` since we don't read it, right?